backend/MangaMagnet.Api/Service/WebSocketService.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Concurrent;
+using System.Net.WebSockets;
+
+namespace MangaMagnet.Api.Service;
+
+public class WebSocketService
+{
+	private readonly ConcurrentDictionary<Guid, WebSocket> _sockets = new();
+
+	/// <summary>
+	/// Add a new socket to the list.
+	/// </summary>
+	/// <param name="id">Unique identifier of the socket.</param>
+	/// <param name="socket">Socket to add.</param>
+	public void AddSocket(Guid id, WebSocket socket)
+	{
+		_sockets.TryAdd(id, socket);
+	}
+
+	/// <summary>
+	/// Remove a socket from the list.
+	/// </summary>
+	/// <param name="id">Unique identifier of the socket.</param>
+	public void RemoveSocket(Guid id)
+	{
+		_sockets.TryRemove(id, out _);
+	}
+
+	/// <summary>
+	/// Send message to all connected clients.
+	/// </summary>
+	/// <param name="memory">Message to send.</param>
+	/// <param name="stoppingToken">Cancellation token.</param>
+	public async Task SendToAllAsync(ReadOnlyMemory<byte> memory, CancellationToken stoppingToken = default)
+	{
+		foreach (var socket in _sockets.Values)
+		{
+			await socket.SendAsync(memory, WebSocketMessageType.Text, true, stoppingToken);
```

Comment:
1. Need to add a `try..catch` around here
2. Maybe send the message in Parallel, if some clients are slow (or being disconnected) all the other clients will get a delayed updated.